server/src/main/java/com/flipkart/varadhi/deployment/LeanDeploymentVerticleDeployer.java

```diff
@@ -0,0 +1,115 @@
+package com.flipkart.varadhi.deployment;
+
+import com.flipkart.varadhi.VerticleDeployer;
+import com.flipkart.varadhi.config.RestOptions;
+import com.flipkart.varadhi.config.ServerConfiguration;
+import com.flipkart.varadhi.entities.Org;
+import com.flipkart.varadhi.entities.Project;
+import com.flipkart.varadhi.entities.Team;
+import com.flipkart.varadhi.exceptions.VaradhiException;
+import com.flipkart.varadhi.spi.db.MetaStoreProvider;
+import com.flipkart.varadhi.spi.services.MessagingStackProvider;
+import com.flipkart.varadhi.web.routes.RouteDefinition;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.vertx.core.Vertx;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+
+@Slf4j
+public class LeanDeploymentVerticleDeployer extends VerticleDeployer {
+    public LeanDeploymentVerticleDeployer(
+            String hostName, Vertx vertx, ServerConfiguration configuration,
+            MessagingStackProvider messagingStackProvider, MetaStoreProvider metaStoreProvider,
+            MeterRegistry meterRegistry
+    ) {
+        super(hostName, vertx, configuration, messagingStackProvider, metaStoreProvider, meterRegistry);
+        validateLeanDeploymentConstraints(configuration.getRestOptions());
```

Comment:
thinking if it would be better to place this in deployVerticle() method as pre deployment check/validation instead of constructor.